openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java

```diff
@@ -160,80 +159,26 @@ int listCount(
-              "ORDER BY <table>.name DESC "
-              + // Pagination ordering by entity fullyQualifiedName or name (when entity does not
-              // have fqn)
-              "LIMIT :limit"
-              + ") last_rows_subquery ORDER BY name",
-      connectionType = MYSQL)
-  @ConnectionAwareSqlQuery(
-      value =
-          "SELECT json FROM ("
-              + "SELECT <table>.name, <table>.json FROM <table> <postgresCond> AND "
-              + "<table>.name < :before "
-              + // Pagination by entity fullyQualifiedName or name (when entity does not have fqn)
-              "ORDER BY <table>.name DESC "
-              + // Pagination ordering by entity fullyQualifiedName or name (when entity does not
-              // have fqn)
-              "LIMIT :limit"
-              + ") last_rows_subquery ORDER BY name",
-      connectionType = POSTGRES)
-  List<String> listBefore(
-      @Define("table") String table,
-      @BindMap Map<String, ?> params,
-      @Define("mysqlCond") String mysqlCond,
-      @Define("postgresCond") String postgresCond,
-      @Bind("limit") int limit,
-      @Bind("before") String before);
-
-  @ConnectionAwareSqlQuery(
-      value =
-          "SELECT <table>.json FROM <table> <mysqlCond> AND "
-              + "<table>.name > :after "
-              + "ORDER BY <table>.name "
-              + "LIMIT :limit",
-      connectionType = MYSQL)
-  @ConnectionAwareSqlQuery(
-      value =
-          "SELECT <table>.json FROM <table> <postgresCond> AND "
-              + "<table>.name > :after "
-              + "ORDER BY <table>.name "
-              + "LIMIT :limit",
-      connectionType = POSTGRES)
-  List<String> listAfter(
-      @Define("table") String table,
-      @BindMap Map<String, ?> params,
-      @Define("mysqlCond") String mysqlCond,
-      @Define("postgresCond") String postgresCond,
-      @Bind("limit") int limit,
-      @Bind("after") String after);
-
-  @ConnectionAwareSqlQuery(
-      value =
-          "SELECT json FROM ("
-              + "SELECT <table>.id, <table>.name, <table>.json FROM <table> <mysqlCond> AND "
+              + "SELECT <table>.name, <table>.id, <table>.json FROM <table> <mysqlCond> AND "
               + "(<table>.name < :beforeName OR (<table>.name = :beforeName AND <table>.id < :beforeId)) "
-              + // Pagination by entity id or name (when entity have same name)
-              "ORDER BY <table>.name DESC, <table>.id  DESC "
-              + // Pagination by entity id or name (when entity have same name)
+              + // Pagination by entity name or id (when entity name same)
+              "ORDER BY <table>.name DESC,<table>.id DESC "
+              + // Pagination by entity name or id (when entity name same)
               "LIMIT :limit"
-              + ") last_rows_subquery ORDER BY name, id",
+              + ") last_rows_subquery ORDER BY name,id",
```

Comment:
is this the right way to do here? what is it going to sort with?

Reply:
is it better for us to use offset and limit instead of encrypted before & after?

Reply:
Using OFFSET like this—SELECT json FROM <table> <cond> ORDER BY name LIMIT :limit OFFSET :offset—can lead to a full table scan, which may cause performance issues as the dataset grows.

In contrast, using a cursor with > after or < before conditions does not always require a full table scan, offering better performance, especially with large datasets.

Reply:
And for the first question , order by name id, will take columns from the inner subquery we dont need to mention it in the outer query